src/cgutils.cpp

```diff
@@ -1397,7 +1397,13 @@ static Value *emit_getfield_knownidx(Value *strct, unsigned idx, jl_datatype_t *
         }
     }
     else if (strct->getType()->isPointerTy()) { // something stack allocated
+#       ifdef LLVM37
+        Value *addr = builder.CreateConstInBoundsGEP2_32(
+            cast<PointerType>(strct->getType()->getScalarType())->getElementType(),
+            strct, 0, idx);
```

Comment:
not sure if this is correct. I thought that it would just be `elty` here, but that causes LLVM to assert out, so I used the type that is checked in the failing assertion... seems to work, but unsightly.